pkg/config/tlscfg/testdata/gen-ssl-conf.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+# Generates the SSL conf files required for generating certificates.
+
+domain_name="$1"
+output_file="$2"
+
+if [[ -z "$domain_name" || -z "$output_file" ]]; then
+    printf "A script to generate SSL configuration files for testing purposes.\n\n"
+    printf "Usage: ssl-conf-gen.sh DOMAIN_NAME OUTPUT_FILE\n\n"
+    printf "Example: ssl-conf-gen.sh example.com ssl.conf\n"
+    return
```

Comment:
should this be `exit -1` instead?

Reply:
`return` was intentional because this script is `source`d, which means it runs within the parent shell. 

The benefit of `source`-ing is that the `set -ex` that you suggested above will also apply in this script so commands in this script are printed to STDOUT and will also trigger an early exit if something fails.

If developers want to call this script separately, it can be run with `source gen-ssl-conf.sh <args...>`.